libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -32,6 +32,16 @@ bool SyntaxChecker::checkSyntax(ASTNode const& _astRoot)
 	return Error::containsOnlyWarnings(m_errors);
 }
 
+void SyntaxChecker::warning(SourceLocation const& _location, string const& _description)
```

Comment:
We should figure out a way to have a helper for this. I think we have this duplicated in every single file almost by now.